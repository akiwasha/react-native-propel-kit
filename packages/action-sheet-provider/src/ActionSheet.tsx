import ModalDialog, {ModalDialogHandle, ModalDialogProps} from '@mgcrea/react-native-modal-dialog';
import React, {forwardRef, ForwardRefRenderFunction, useMemo} from 'react';
import {
  ActionSheetIOSOptions,
  Dimensions,
  Platform,
  ScrollView,
  ScrollViewProps,
  StyleProp,
  StyleSheet,
  TextStyle,
  ViewStyle
} from 'react-native';
import ActionSheetOption from './components/ActionSheetOption';
import isUndefined from './utils/isUndefined';

const {height: WINDOW_HEIGHT} = Dimensions.get('window');
const IOS_OPACITY = 0.85; // @NOTE from native ActionSheet (though it uses blur)

export type ActionSheetProps = Pick<ScrollViewProps, 'scrollEnabled'> &
  Omit<ModalDialogProps, 'onConfirm' | 'confirmTitle' | 'confirmStyle'> &
  ActionSheetIOSOptions & {
    defaultStyles?: typeof defaultStyles;
    optionStyle?: StyleProp<TextStyle>;
    lastOptionExtraStyle?: StyleProp<TextStyle>;
    destructiveButtonColor?: string;
    onButtonPress: (buttonIndex: number) => void;
    onCancel: () => void;
  };

export const defaultProps = {
  backgroundColor: 'white',
  cancelTitle: 'Cancel',
  destructiveButtonColor: Platform.select({
    ios: '#ff3b30', // iOS.systemRed (@see https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/color/)
    android: '#f44336' // android.red500 (@see https://material.io/design/color/the-color-system.html)
  })!
};

const ActionSheetRenderFunction: ForwardRefRenderFunction<ModalDialogHandle, ActionSheetProps> = (
  {
    options,
    backgroundColor = defaultProps.backgroundColor,
    optionStyle: propOptionStyle,
    lastOptionExtraStyle: propLastOptionExtraStyle,
    cancelButtonIndex,
    destructiveButtonIndex,
    destructiveButtonColor = defaultProps.destructiveButtonColor,
    onButtonPress,
    scrollEnabled: propScrollEnabled,
    defaultStyles: propDefaultStyles = defaultStyles,
    ...otherModalDialogProps
  },
  ref
) => {
  const cancelTitle = !isUndefined(cancelButtonIndex) && options[cancelButtonIndex] ? options[cancelButtonIndex] : '';
  const scrollEnabled = !isUndefined(cancelButtonIndex) ? options.length > 6 : propScrollEnabled;

  const optionStyle = useMemo<ViewStyle>(
    () => StyleSheet.flatten([propDefaultStyles.option, {backgroundColor}, propOptionStyle]),
    [propDefaultStyles, backgroundColor, propOptionStyle]
  );
  const lastOptionExtraStyle = useMemo<ViewStyle>(
    () => StyleSheet.flatten([propDefaultStyles.lastOptionExtra, propLastOptionExtraStyle]),
    [propDefaultStyles, propLastOptionExtraStyle]
  );

  return (
    <ModalDialog
      ref={ref}
      defaultStyles={propDefaultStyles}
      backgroundColor={backgroundColor}
      cancelTitle={cancelTitle}
      bodyStyle={{backgroundColor: 'transparent'}}
      {...otherModalDialogProps}>
      <ScrollView scrollEnabled={scrollEnabled}>
        {options.map((buttonTitle, buttonIndex) => {
          // Skip cancel button
          if (!isUndefined(cancelButtonIndex) && buttonIndex === cancelButtonIndex) {
            return null;
          }

          const color: string =
            destructiveButtonIndex && buttonIndex === destructiveButtonIndex ? destructiveButtonColor : '';
          return (
            <ActionSheetOption
              style={[optionStyle as TextStyle].concat(
                color ? [{color}] : [],
                buttonIndex === options.length - 1 ? [lastOptionExtraStyle] : []
              )}
              onPress={() => onButtonPress(buttonIndex)}
              title={buttonTitle}
              key={`${buttonIndex}`} // eslint-disable-line react/no-array-index-key
            />
          );
        })}
      </ScrollView>
    </ModalDialog>
  );
};

export const ActionSheet = forwardRef(ActionSheetRenderFunction);

export const defaultStyles = {
  modal: Platform.select<ViewStyle>({
    ios: {flex: 1, justifyContent: 'flex-end'},
    android: {flex: 1, justifyContent: 'center'}
  })!,
  container: Platform.select<ViewStyle>({
    ios: {
      flexDirection: 'column',
      alignItems: 'stretch',
      marginHorizontal: 8,
      marginBottom: 8,
      maxHeight: WINDOW_HEIGHT - 8
    },
    android: {
      flexDirection: 'column',
      alignItems: 'stretch',
      // marginHorizontal: 82, // timepicker/spinner
      margin: 24,
      maxHeight: WINDOW_HEIGHT - 24 * 2,
      elevation: 4
    }
  })!,
  header: Platform.select<ViewStyle>({
    ios: {
      alignItems: 'center',
      padding: 16,
      borderTopLeftRadius: 12,
      borderTopRightRadius: 12,
      opacity: IOS_OPACITY
    },
    android: {alignItems: 'flex-start', paddingVertical: 24, paddingHorizontal: 32}
  })!,
  body: Platform.select<ViewStyle>({
    ios: {
      flexShrink: 1,
      flexDirection: 'column',
      backgroundColor: 'transparent',
      opacity: IOS_OPACITY,
      borderBottomStartRadius: 12,
      borderBottomEndRadius: 12
    },
    android: {
      flexShrink: 1,
      flexDirection: 'column',
      backgroundColor: 'transparent'
    }
  })!,
  footer: Platform.select<TextStyle>({
    ios: {}, // @NOTE we don't have a footer on iOS
    android: {
      flexDirection: 'row',
      justifyContent: 'flex-end',
      borderBottomLeftRadius: 2,
      borderBottomRightRadius: 2,
      padding: 8
    }
  })!,
  title: Platform.select<TextStyle>({
    ios: {paddingBottom: 12, fontSize: 14, fontWeight: '500', textAlign: 'center', color: '#888'},
    android: {
      paddingBottom: 12,
      fontSize: 20,
      fontWeight: '500',
      textAlign: 'left',
      color: '#333'
    }
  })!,
  message: Platform.select<TextStyle>({
    ios: {paddingBottom: 12, fontSize: 13, fontWeight: '400', textAlign: 'center', color: '#888'},
    android: {paddingBottom: 12, fontSize: 18, fontWeight: '400', color: '#666'}
  })!,
  cancel: Platform.select<TextStyle>({
    ios: {marginTop: 8, borderRadius: 12, fontWeight: '600'},
    android: {}
  })!,
  confirm: Platform.select<TextStyle>({
    ios: {}, // @NOTE we don't have a confirm on iOS
    android: {}
  })!,
  option: Platform.select<TextStyle>({
    ios: {
      marginTop: StyleSheet.hairlineWidth
    },
    android: {
      marginTop: StyleSheet.hairlineWidth
    }
  })!,
  lastOptionExtra: Platform.select<TextStyle>({
    ios: {
      borderBottomLeftRadius: 12,
      borderBottomRightRadius: 12
    },
    android: {
      marginBottom: StyleSheet.hairlineWidth
    }
  })!
};
