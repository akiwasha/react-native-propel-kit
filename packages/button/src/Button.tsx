// @see https://github.com/facebook/react-native/blob/master/Libraries/Components/Button.js
// @see https://github.com/react-native-training/react-native-elements/blob/next/src/buttons/Button.js

import React, {ElementType, FunctionComponent, useCallback, useMemo} from 'react';
import {
  ButtonProps as BaseButtonProps,
  GestureResponderEvent,
  Platform,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  TouchableNativeFeedback,
  TouchableNativeFeedbackProps,
  TouchableOpacity,
  TouchableOpacityProps,
  View,
  ViewStyle
} from 'react-native';
import {pickTextStyles} from './utils';

export const defaultStyles = {
  view: Platform.select<ViewStyle>({
    ios: {justifyContent: 'center', flexDirection: 'row', height: 50},
    android: {justifyContent: 'center', flexDirection: 'row', height: 50}
  }),
  disabledView: Platform.select<ViewStyle>({
    ios: {},
    android: {}
  }),
  text: Platform.select<TextStyle>({
    ios: {textAlign: 'center', margin: 8, color: '#007aff', fontSize: 18},
    android: {paddingHorizontal: 3, paddingBottom: 2, color: '#212121'}
  }),
  disabledText: Platform.select<TextStyle>({
    ios: {color: '#cdcdcd'},
    android: {color: '#cdcdcd'}
  })
  // loading: Platform.select<ViewStyle>({
  //   ios: {
  //     position: 'absolute',
  //     backgroundColor: 'black',
  //     opacity: 0.5,
  //     height: 40,
  //     width: 40,
  //     marginRight: 8,
  //     alignSelf: 'center',
  //     borderRadius: 12
  //   },
  //   android: {marginRight: 4}
  // })
};

export type ButtonProps = BaseButtonProps & {
  // loading?: boolean;
  // loadingStyle?: StyleProp<ViewStyle>;
  style?: StyleProp<TextStyle>;
  viewStyle?: StyleProp<ViewStyle>;
  disabledViewStyle?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  disabledTextStyle?: StyleProp<TextStyle>;
  TouchableComponent?: ElementType<TouchableNativeFeedbackProps | TouchableOpacityProps>;
  activeOpacity?: TouchableOpacityProps['activeOpacity'];
  defaultStyles?: typeof defaultStyles;
};

export const defaultProps = {
  TouchableComponent: Platform.select<ElementType<TouchableNativeFeedbackProps | TouchableOpacityProps>>({
    android: TouchableNativeFeedback,
    default: TouchableOpacity
  }),
  defaultStyles: typeof defaultStyles
};

export const Button: FunctionComponent<ButtonProps> = ({
  title,
  disabled,
  // loading: propLoading,
  onPress,
  color,
  viewStyle: propViewStyle,
  // loadingStyle: propLoadingStyle,
  disabledViewStyle,
  textStyle: propTextStyle,
  disabledTextStyle,
  defaultStyles: propDefaultStyles = defaultStyles,
  style: propStyle,
  TouchableComponent = defaultProps.TouchableComponent,
  ...otherTouchableProps
}) => {
  // @NOTE we use flatten to properly split text/view related styles
  const flattenStyle = useMemo<TextStyle>(() => (propStyle ? StyleSheet.flatten(propStyle) : {}), [propStyle]);

  const handlePress = useCallback(
    (ev: GestureResponderEvent) => {
      if (disabled) {
        return;
      }
      if (onPress) {
        onPress(ev);
      }
    },
    [disabled, onPress]
  );

  const viewStyle = useMemo<ViewStyle>(
    () =>
      StyleSheet.flatten([
        propDefaultStyles.view,
        propViewStyle,
        flattenStyle,
        disabled ? [propDefaultStyles.disabledView, disabledViewStyle] : null
      ]),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [propViewStyle, flattenStyle, disabled, disabledViewStyle]
  );

  const textStyle = useMemo<TextStyle>(
    () =>
      StyleSheet.flatten([
        propDefaultStyles.text,
        propTextStyle,
        pickTextStyles(flattenStyle),
        color ? {color} : null,
        disabled ? [propDefaultStyles.disabledText, disabledTextStyle] : null
      ]),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [propTextStyle, flattenStyle, disabled, color, disabledTextStyle]
  );

  return (
    <TouchableComponent
      style={{flex: viewStyle.flex ? 1 : 0, flexDirection: 'row'}}
      disabled={disabled}
      onPress={handlePress}
      {...otherTouchableProps}
    >
      <View style={viewStyle}>
        {/* {propLoading ? (
          <ActivityIndicator
            style={[propDefaultStyles.loading, propLoadingStyle]}
            color={textStyle && textStyle.color ? textStyle.color : propDefaultStyles.text.color}
          />
        ) : null} */}
        <Text style={textStyle}>{title}</Text>
      </View>
    </TouchableComponent>
  );
};
