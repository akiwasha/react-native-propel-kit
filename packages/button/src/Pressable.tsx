// @see https://github.com/facebook/react-native/blob/master/Libraries/Components/Button.js
// @see https://github.com/react-native-training/react-native-elements/blob/next/src/buttons/Button.js

import React, {ElementType, FunctionComponent, ReactNode, useCallback, useMemo} from 'react';
import {
  ButtonProps,
  GestureResponderEvent,
  Platform,
  StyleProp,
  StyleSheet,
  Text,
  TextProps,
  TextStyle,
  TouchableNativeFeedback,
  TouchableNativeFeedbackProps,
  TouchableOpacity,
  TouchableOpacityProps,
  ViewStyle
} from 'react-native';
import {pickTextStyles} from './utils';

export type PressableProps = ButtonProps &
  Pick<TextProps, 'ellipsizeMode' | 'numberOfLines'> & {
    children?: ReactNode;
    style?: StyleProp<TextStyle>;
    viewStyle?: StyleProp<ViewStyle>;
    disabledViewStyle?: StyleProp<ViewStyle>;
    textStyle?: StyleProp<TextStyle>;
    disabledTextStyle?: StyleProp<TextStyle>;
    TouchableComponent?: ElementType<TouchableNativeFeedbackProps | TouchableOpacityProps>;
    activeOpacity?: TouchableOpacityProps['activeOpacity'];
    defaultStyles?: typeof defaultStyles;
  };

export const defaultProps = {
  TouchableComponent: Platform.select<ElementType<TouchableNativeFeedbackProps | TouchableOpacityProps>>({
    android: TouchableNativeFeedback,
    default: TouchableOpacity
  })
};

export const Pressable: FunctionComponent<PressableProps> = ({
  children,
  title,
  disabled,
  onPress,
  color,
  viewStyle: propViewStyle,
  disabledViewStyle,
  textStyle: propTextStyle,
  ellipsizeMode,
  numberOfLines,
  disabledTextStyle,
  defaultStyles: propDefaultStyles = defaultStyles,
  style: propStyle,
  TouchableComponent = defaultProps.TouchableComponent,
  ...otherTouchableProps
}) => {
  // @NOTE we use flatten to properly split text/view related styles
  const flattenStyle = useMemo<TextStyle>(() => (propStyle ? StyleSheet.flatten(propStyle) : {}), [propStyle]);

  const handlePress = useCallback(
    (ev: GestureResponderEvent) => {
      if (disabled) {
        return;
      }
      if (onPress) {
        onPress(ev);
      }
    },
    [disabled, onPress]
  );

  const viewStyle = useMemo<ViewStyle>(
    () =>
      StyleSheet.flatten([propDefaultStyles.view, propViewStyle, flattenStyle, disabled ? disabledViewStyle : null]),
    [propDefaultStyles.view, propViewStyle, flattenStyle, disabled, disabledViewStyle]
  );

  const textStyle = useMemo<TextStyle>(
    () =>
      StyleSheet.flatten([
        propDefaultStyles.text,
        propTextStyle,
        pickTextStyles(flattenStyle),
        color ? {color} : null,
        disabled ? disabledTextStyle : null
      ]),
    [propDefaultStyles.text, propTextStyle, flattenStyle, disabled, color, disabledTextStyle]
  );

  return (
    <TouchableComponent style={viewStyle} disabled={disabled} onPress={handlePress} {...otherTouchableProps}>
      {children}
      <Text ellipsizeMode={ellipsizeMode} numberOfLines={numberOfLines} style={textStyle}>
        {title}
      </Text>
    </TouchableComponent>
  );
};

export const defaultStyles = {
  view: Platform.select<ViewStyle>({
    ios: {justifyContent: 'center', flexDirection: 'row', height: 50},
    android: {justifyContent: 'center', flexDirection: 'row', height: 50}
  }),
  disabledView: Platform.select<ViewStyle>({
    ios: {},
    android: {}
  }),
  text: Platform.select<TextStyle>({
    ios: {alignSelf: 'center', fontSize: 18, color: '#007aff'},
    android: {paddingHorizontal: 3, paddingBottom: 2, color: '#212121'}
  }),
  disabledText: Platform.select<TextStyle>({
    ios: {color: '#cdcdcd'},
    android: {color: '#cdcdcd'}
  })
};
