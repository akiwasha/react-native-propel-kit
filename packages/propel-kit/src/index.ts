export * from '@mgcrea/react-native-action-sheet-provider';
export * from '@mgcrea/react-native-backdrop-provider';
export * from '@mgcrea/react-native-button';
export * from '@mgcrea/react-native-date-picker';
export * from '@mgcrea/react-native-modal-dialog';
export * from '@mgcrea/react-native-select';
