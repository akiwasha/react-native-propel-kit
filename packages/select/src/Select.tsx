import {InputButton, InputButtonProps} from '@mgcrea/react-native-button';
import ModalDialog, {ModalDialogHandle, ModalDialogProps} from '@mgcrea/react-native-modal-dialog';
import {Picker} from '@react-native-picker/picker';
import React, {
  Children,
  ElementType,
  forwardRef,
  ReactElement,
  ReactNode,
  useCallback,
  useEffect,
  useImperativeHandle,
  useMemo,
  useRef,
  useState
} from 'react';
import {StyleProp, TextStyle} from 'react-native';

const isUndefined = (maybeUndefined: any): maybeUndefined is undefined => typeof maybeUndefined === 'undefined';

export const SelectItem = Picker.Item;

export type SelectValue = string | number | undefined;

export type SelectHandle = {
  focus: () => void;
};

export type SelectProps = Pick<ModalDialogProps, 'title' | 'confirmTitle' | 'cancelTitle'> & {
  children?: ReactNode;
  initialValue?: SelectValue;
  InputButtonComponent?: ElementType<InputButtonProps>;
  onChange?: (value: SelectValue) => void;
  onEndEditing?: () => void;
  onSubmitEditing?: (value: SelectValue) => void;
  placeholder?: string;
  style?: StyleProp<TextStyle>;
  value?: SelectValue;
  [s: string]: unknown; // unknown otherProps for InputButtonComponent
};

export const defaultProps = {
  InputButtonComponent: InputButton
};

export const Select = Object.assign(
  forwardRef<SelectHandle, SelectProps>(
    (
      {
        children,
        // ModalDialog props
        cancelTitle,
        confirmTitle,
        title,
        // Select props
        initialValue: propInitialValue,
        InputButtonComponent = defaultProps.InputButtonComponent,
        onChange: propOnChange,
        onEndEditing,
        onSubmitEditing,
        placeholder,
        value: propValue,
        ...otherProps
      },
      ref
    ) => {
      const modalDialogRef = useRef<ModalDialogHandle>(null);
      const [localValue, setLocalValue] = useState<SelectValue>();
      const [modalValue, setModalValue] = useState<SelectValue>(propValue || propInitialValue);

      // Support both controlled/uncontrolled usage
      const inputValue = useMemo(() => (!isUndefined(propValue) ? propValue : localValue), [propValue, localValue]);

      // Track parent propValue controlled updates
      useEffect(() => {
        setModalValue(propValue || propInitialValue);
      }, [propValue, propInitialValue]);

      // Lazily compute displayed label
      const labelValue = useMemo(() => {
        if (isUndefined(inputValue)) {
          return '';
        }
        const childrenAsArray: ReactElement[] = React.Children.toArray(children) as ReactElement[];
        const selectedChild = childrenAsArray.find((child) => child.props.value === inputValue);
        const label = selectedChild ? selectedChild.props.label || selectedChild.props.value : '';
        return label;
      }, [children, inputValue]);

      // Propagate changed value
      const onConfirm = useCallback(() => {
        if (propOnChange) {
          propOnChange(modalValue);
        }
        if (onEndEditing) {
          onEndEditing();
        }
        if (onSubmitEditing) {
          // @NOTE Add a timeout to prevent swallowing siblings focus events
          setTimeout(() => {
            onSubmitEditing(modalValue);
          });
        }
        // Support uncontrolled usage
        setLocalValue(modalValue);
      }, [modalValue, onSubmitEditing, onEndEditing, propOnChange]);

      // Reset modalValue to the proper value
      const onCancel = useCallback(() => {
        setModalValue(inputValue || propInitialValue);
      }, [inputValue, propInitialValue]);

      // Open the modal when user touches input
      const focus = useCallback(() => {
        if (modalDialogRef.current) {
          modalDialogRef.current.show();
        }
      }, []);

      // Expose API via an imperative handle
      useImperativeHandle(ref, () => ({focus}), [focus]);

      // @NOTE Hack around lousy handling of picker children (@see https://github.com/facebook/react-native/pull/8153)
      const pickerChildren = useMemo<ReactNode | ReactNode[]>(
        () =>
          propInitialValue
            ? children
            : ([<Picker.Item key="undefined" label="" value={undefined} />] as ReactNode[]).concat(
                Children.toArray(children)
              ),
        [propInitialValue, children]
      );

      // Track modal value updates
      const onValueChange = useCallback((value) => {
        setModalValue(value);
      }, []);

      return (
        <>
          <InputButtonComponent onFocus={focus} placeholder={placeholder} value={labelValue} {...otherProps} />
          <ModalDialog
            ref={modalDialogRef}
            title={title || placeholder}
            onConfirm={onConfirm}
            onCancel={onCancel}
            confirmTitle={confirmTitle}
            cancelTitle={cancelTitle}>
            <Picker style={{flexGrow: 1}} onValueChange={onValueChange} selectedValue={modalValue}>
              {pickerChildren}
            </Picker>
          </ModalDialog>
        </>
      );
    }
  ),
  {Item: SelectItem}
);
