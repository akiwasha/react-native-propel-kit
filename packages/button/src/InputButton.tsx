import React, {FunctionComponent} from 'react';
import {Platform, StyleProp, TextInputProps, TextStyle, ViewStyle} from 'react-native';
import {Pressable, PressableProps} from './Pressable';

export type InputButtonProps = Omit<TextInputProps, 'onPress'> & {
  onFocus: PressableProps['onPress'];
  viewStyle?: StyleProp<ViewStyle>;
  disabledViewStyle?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  disabledTextStyle?: StyleProp<TextStyle>;
};

export const defaultProps = {
  placeholderTextColor: Platform.select({ios: '#c7c7cc', android: '#9e9e9e'})
};

export const InputButton: FunctionComponent<InputButtonProps> = ({
  editable = true,
  onFocus,
  placeholder,
  viewStyle,
  disabledViewStyle,
  textStyle,
  disabledTextStyle,
  placeholderTextColor = defaultProps.placeholderTextColor,
  value,
  ...otherPressableProps
}) => {
  return (
    <Pressable
      onPress={onFocus}
      title={value || placeholder || ''}
      disabled={!editable}
      viewStyle={[defaultStyles.view, viewStyle]}
      disabledViewStyle={[defaultStyles.disabledView, disabledViewStyle]}
      textStyle={[defaultStyles.text, textStyle]}
      disabledTextStyle={[defaultStyles.disabledText, disabledTextStyle]}
      color={!value ? placeholderTextColor : undefined}
      numberOfLines={1}
      {...otherPressableProps}
    />
  );
};

export const defaultStyles: {[s: string]: ViewStyle | TextStyle} = {
  view: Platform.select<ViewStyle>({
    ios: {justifyContent: 'flex-start', alignItems: 'center'},
    android: {justifyContent: 'flex-start', height: 50}
  })!,
  disabledView: Platform.select<ViewStyle>({
    ios: {},
    android: {}
  })!,
  text: Platform.select<TextStyle>({
    ios: {fontSize: 14, color: 'black'},
    android: {alignSelf: 'flex-start', paddingHorizontal: 3, paddingBottom: 2, color: '#212121'}
  })!,
  disabledText: Platform.select<TextStyle>({
    ios: {color: '#cdcdcd'},
    android: {color: '#cdcdcd'}
  })!
};
