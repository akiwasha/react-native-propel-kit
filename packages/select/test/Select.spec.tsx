import React from 'react';
import 'react-native';
// import renderer from 'react-test-renderer';
import {render, act, fireEvent} from 'react-native-testing-library';

import {Select} from '../src';

describe('<Select>', () => {
  it('renders correctly', () => {
    const component = render(<Select />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
