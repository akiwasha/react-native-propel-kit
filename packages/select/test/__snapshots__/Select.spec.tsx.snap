// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select> renders correctly 1`] = `
Array [
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 0,
        "flexDirection": "row",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "alignSelf": "center",
            "color": "#c7c7cc",
            "fontSize": 14,
          }
        }
      >
        
      </Text>
    </View>
  </View>,
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
          },
          undefined,
        ]
      }
    >
      <RCTSafeAreaView
        accessible={true}
        emulateUnlessSupported={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "stretch",
              "flexDirection": "column",
              "marginBottom": 8,
              "marginHorizontal": 8,
              "maxHeight": 1326,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "column",
                "flexShrink": 1,
              },
              Object {
                "borderTopLeftRadius": 12,
                "borderTopRightRadius": 12,
              },
              Object {
                "backgroundColor": "white",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <RNCPicker
              items={
                Array [
                  Object {
                    "label": "",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": undefined,
                  },
                ]
              }
              numberOfLines={1}
              onChange={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "borderBottomLeftRadius": 12,
                "borderBottomRightRadius": 12,
              },
              Object {
                "backgroundColor": "white",
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 0,
                "flexDirection": "row",
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomLeftRadius": 12,
                  "borderBottomRightRadius": 12,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "fontWeight": "400",
                  "height": 56,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#007aff",
                    "fontSize": 20,
                    "fontWeight": "400",
                  }
                }
              >
                Confirm
              </Text>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 0,
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderRadius": 12,
                "flexDirection": "row",
                "flexGrow": 1,
                "fontWeight": "600",
                "height": 56,
                "justifyContent": "center",
                "marginTop": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#007aff",
                  "fontSize": 20,
                  "fontWeight": "600",
                }
              }
            >
              Cancel
            </Text>
          </View>
        </View>
      </RCTSafeAreaView>
    </View>
  </Modal>,
]
`;
